### PR TITLE
Add IPv6 support to CSDK

### PR DESCRIPTION
**Pull request by shrikant1407**
* Add IPv6 support to CSDK
* Add curl support for HTTP connection

Signed-off-by: Shrikant Temburwar <shrikant.temburwar@intel.com>

### CODE REVIEW

File: network/network_if_linux.c [hunk 1]
```diff
@@ -67,21 +67,17 @@ struct fdo_sock_handle {
 /**
  * Read from socket until new-line is encountered.
  *
```
Comment (tajnisha) [inline network/network_if_linux.c:69]: Update this comments to update with curl buf instead of socket

File: lib/fdoprotctx.c [hunk 2]
```diff
@@ -412,7 +409,7 @@ int fdo_prot_ctx_run(fdo_prot_ctx_t *prot_ctx)
 			}
 		}
 
-		if (fdo_con_disconnect(prot_ctx->sock_hdl, prot_ctx->tls)) {
+		if (fdo_con_disconnect(prot_ctx->sock_hdl)) {
 			LOG(LOG_ERROR, "Error during socket close()\n");
```
Comment (tajnisha) [inline lib/fdoprotctx.c:413]: Update the log messages to remove socket close

File: network/network_if_linux.c [hunk 3]
```diff
@@ -365,43 +390,56 @@ int fdo_curl_setup(fdo_ip_address_t *ip_addr, uint16_t port)
...
 
 	if (curl) {
-		// we are directed to enforce TLS
-		curl_version_info_data * vinfo = curl_version_info(CURLVERSION_NOW);
-		if (CURL_VERSION_SSL == (vinfo->features & CURL_VERSION_SSL)) {
-			// SSL support enabled
-			LOG(LOG_INFO, "SSL support verified.\n");
+		//TO-DO: Remove this in prod code
+		if (is_ipv6) {
+			if (curl_easy_setopt(curl, CURLOPT_PROXY, "") != CURLE_OK) {
```
Comment (tajnisha) [inline network/network_if_linux.c:395]: We will discuss more on this.

File: network/network_if_linux.c [hunk 4]
```diff
@@ -365,43 +390,56 @@ int fdo_curl_setup(fdo_ip_address_t *ip_addr, uint16_t port)
...
+				LOG(LOG_ERROR, "CURL_PROXY: Cannot set proxy.\n");
+				goto err;
+			}
+			if (curl_easy_setopt(curl, CURLOPT_INTERFACE, "ens18") != CURLE_OK) {
```
[the remark this answers is not in the document]
Comment (shrikant1407) [reply network/network_if_linux.c]: added runtime option

**Pull request closed without merging**